buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersPlugin.java

```diff
@@ -293,16 +304,56 @@ private static void autoConfigureClusterDependencies(
         // We need afterEvaluate here despite the fact that container is a domain object, we can't implement this with
         // all because fields can change after the fact.
         project.afterEvaluate(ip -> container.forEach(esNode -> {
-            // declare dependencies against artifacts needed by cluster formation.
-            String dependency = String.format(
-                "unused:%s:%s:%s@%s",
-                esNode.getDistribution().getArtifactName(),
-                esNode.getVersion(),
-                esNode.getDistribution().getClassifier(),
-                esNode.getDistribution().getFileExtension()
-            );
-            logger.info("Cluster {} depends on {}", esNode.getName(), dependency);
-            rootProject.getDependencies().add(HELPER_CONFIGURATION_NAME, dependency);
+            final VersionCollection bwcVersions;
+            final List<Version> unreleased;
+            {
```

Comment:
Why is this code nested in a block?

Reply:
It was supposed to be for scoping of the `bwcVersions` variable but I lost track of that somehow. I fixed it now, it's to make it intentionally hard to use this to determine dependencies.